### ISSUE TITLE
Stored Object: paper:arxiv.2107.09133

### ISSUE THREAD

**Issue by dmarx** [dmarx/papers-feed-dev#20]
{
  "primary_id": "arxiv.2107.09133",
  "source": "arxiv",
  "sourceId": "2107.09133",
  "url": "https://arxiv.org/abs/2107.09133",
  "title": "The Limiting Dynamics of SGD: Modified Loss, Phase Space Oscillations,\n  and Anomalous Diffusion",
  "authors": "Daniel Kunin, Javier Sagastuy-Brena, Lauren Gillespie, Eshed Margalit, Hidenori Tanaka, Surya Ganguli, Daniel L. K. Yamins",
  "abstract": "In this work we explore the limiting dynamics of deep neural networks trained\nwith stochastic gradient descent (SGD). As observed previously, long after\nperformance has converged, networks continue to move through parameter space by\na process of anomalous diffusion in which distance travelled grows as a power\nlaw in the number of gradient updates with a nontrivial exponent. We reveal an\nintricate interaction between the hyperparameters of optimization, the\nstructure in the gradient noise, and the Hessian matrix at the end of training\nthat explains this anomalous diffusion. To build this understanding, we first\nderive a continuous-time model for SGD with finite learning rates and batch\nsizes as an underdamped Langevin equation. We study this equation in the\nsetting of linear regression, where we can derive exact, analytic expressions\nfor the phase space dynamics of the parameters and their instantaneous\nvelocities from initialization to stationarity. Using the Fokker-Planck\nequation, we show that the key ingredient driving these dynamics is not the\noriginal training loss, but rather the combination of a modified loss, which\nimplicitly regularizes the velocity, and probability currents, which cause\noscillations in phase space. We identify qualitative and quantitative\npredictions of this theory in the dynamics of a ResNet-18 model trained on\nImageNet. Through the lens of statistical physics, we uncover a mechanistic\norigin for the anomalous limiting dynamics of deep neural networks trained with\nSGD.",
  "timestamp": "2025-03-01T20:28:38.002Z",
  "rating": "novote",
  "arxivId": "2107.09133",
  "arxiv_tags": [
    "cs.LG",
    "cond-mat.stat-mech",
    "q-bio.NC",
    "stat.ML"
  ],
  "published_date": "2021-07-19T20:18:57Z"
}

---
*Copied from dmarx/papers-feed#1536*

**Comment by dmarx** [comment]
{
  "_data": {
    "primary_id": "arxiv.2107.09133",
    "source": "arxiv",
    "sourceId": "2107.09133",
    "url": "https://arxiv.org/abs/2107.09133",
    "title": "The Limiting Dynamics of SGD: Modified Loss, Phase Space Oscillations,\n  and Anomalous Diffusion",
    "authors": "Daniel Kunin, Javier Sagastuy-Brena, Lauren Gillespie, Eshed Margalit, Hidenori Tanaka, Surya Ganguli, Daniel L. K. Yamins",
    "abstract": "In this work we explore the limiting dynamics of deep neural networks trained\nwith stochastic gradient descent (SGD). As observed previously, long after\nperformance has converged, networks continue to move through parameter space by\na process of anomalous diffusion in which distance travelled grows as a power\nlaw in the number of gradient updates with a nontrivial exponent. We reveal an\nintricate interaction between the hyperparameters of optimization, the\nstructure in the gradient noise, and the Hessian matrix at the end of training\nthat explains this anomalous diffusion. To build this understanding, we first\nderive a continuous-time model for SGD with finite learning rates and batch\nsizes as an underdamped Langevin equation. We study this equation in the\nsetting of linear regression, where we can derive exact, analytic expressions\nfor the phase space dynamics of the parameters and their instantaneous\nvelocities from initialization to stationarity. Using the Fokker-Planck\nequation, we show that the key ingredient driving these dynamics is not the\noriginal training loss, but rather the combination of a modified loss, which\nimplicitly regularizes the velocity, and probability currents, which cause\noscillations in phase space. We identify qualitative and quantitative\npredictions of this theory in the dynamics of a ResNet-18 model trained on\nImageNet. Through the lens of statistical physics, we uncover a mechanistic\norigin for the anomalous limiting dynamics of deep neural networks trained with\nSGD.",
    "timestamp": "2025-03-01T20:28:38.002Z",
    "rating": "novote",
    "arxivId": "2107.09133",
    "arxiv_tags": [
      "cs.LG",
      "cond-mat.stat-mech",
      "q-bio.NC",
      "stat.ML"
    ],
    "published_date": "2021-07-19T20:18:57Z"
  },
  "_meta": {
    "client_version": "0.3.2",
    "timestamp": "2025-03-01T20:28:38.947Z",
    "update_mode": "append"
  },
  "type": "initial_state"
}

---
*Original comment by @dmarx*
